wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/SynapseTestCaseFileReader.java

```diff
@@ -0,0 +1,157 @@
+
+        //Read test-artifact data
+        QName qualifiedArtifact = new QName("", Constants.ARTIFACT, "");
+        OMElement testArtifactFileNode = testArtifactNode.getFirstChildWithName(qualifiedArtifact);
+
+        String testArtifactFileAsString;
+        if (!testArtifactFileNode.getText().isEmpty()) {
+            String testArtifactFilePath = testArtifactFileNode.getText();
+            testArtifactFileAsString = FileUtils.readFileToString(new File(testArtifactFilePath));
+        } else {
+            throw new IOException("Test artifact does not contain configuration file path");
+        }
+
+        if (testArtifactFileAsString != null) {
+            //add test-artifact data as a child
+            OMElement testArtifactDataNode = AXIOMUtil.stringToOM(testArtifactFileAsString);
+            testArtifactFileNode.removeChildren();
+            testArtifactFileNode.addChild(testArtifactDataNode);
+        } else {
+            throw new IOException("Test artifact does not contain any configuration data");
+        }
+    }
+
+    /**
+     * Method of processing supportive-artifact data.
+     * Reads artifacts from user defined file and append it to the artifact node
+     *
+     * @param artifactsNode artifact data contain node
+     */
+    private static void processSupportiveArtifactData(OMElement artifactsNode) throws IOException, XMLStreamException {
+        QName qualifiedSupportiveArtifact = new QName("", Constants.SUPPORTIVE_ARTIFACTS, "");
+        OMElement supportiveArtifactNode = artifactsNode.getFirstChildWithName(qualifiedSupportiveArtifact);
+
+        Iterator artifactIterator = Collections.emptyIterator();
+        if (supportiveArtifactNode != null) {
+            artifactIterator = supportiveArtifactNode.getChildElements();
+        }
+
+        while (artifactIterator.hasNext()) {
+            OMElement artifact = (OMElement) artifactIterator.next();
+
+            String supportiveArtifactFileAsString;
+            if (!artifact.getText().isEmpty()) {
+                String artifactFilePath = artifact.getText();
+                supportiveArtifactFileAsString = FileUtils.readFileToString(new File(artifactFilePath));
+            } else {
+                throw new IOException("Supportive artifact does not contain configuration file path");
+            }
+
+            if (supportiveArtifactFileAsString != null) {
+                //add test-artifact data as a child
+                OMElement supportiveArtifactDataNode = AXIOMUtil.stringToOM(supportiveArtifactFileAsString);
+                artifact.removeChildren();
+                artifact.addChild(supportiveArtifactDataNode);
+            } else {
+                throw new IOException("Supportive artifact does not contain any configuration data");
+            }
+        }
+    }
+
+    /**
+     * Method of initiating logger.
+     */
+    private static Log getLog() {
```

Comment:
Can we do this log printing same as using LogFactory(as following example) or is it implemented in this way for this repository? 

private static final Log log = LogFactory.getLog(SynapseTestCaseFileReader.class);
log.info("bla");

Reply:
I used this log because of this is the log type used in AbstractMojo class using the org.apache.maven.plugin.logging.Log library. (to keep the log flow same as others while maven build).  Is it need to change?

Reply:
If this is the standard way for this repo, then it's fine. normally we are logging by log.info("bla") format and it is common for most of the WSO2 repos. I though we can create logging class same as LogFactory for this repo as well.